docs/design/kademlia-removal.md

```diff
@@ -0,0 +1,96 @@
+# Kademlia Removal
+
+## Abstract
+
+This design document outlines the communication protocol between satellites and
+storage nodes, network refreshes, and kademlia removal given the satellite opt-in
+capability for storage nodes.
+
+## Background
+
+Many peer-to-peer, decentralized systems employ the Kademlia implementation of a distributed hash table to allow for 
+locating peer nodes, exchanging messages and sharing data. However, due to the nature of our network, we only use Kademlia 
+for node discovery and address lookups given node IDs. This is useful when satellites don’t know about all the nodes in 
+the network and nodes are unfamiliar with all of the satellites in the network.
+
+With our recent business decision of simplifying the storage node operator user experience, we no longer require kademlia 
+for node discovery. In a solution called SNO-select, storage nodes manually select the satellites they want to work with 
+and satellites wait for storage nodes to work with them. The initial implementation of this solution allows SNOs to update 
+their trusted satellite list in a hardcoded configuration file, but future improvements will enable users to manage
+this list through a web console. 
+
+We will replace our Kademlia DHT and related entities with direct communication between satellites and storage nodes, 
+and keep the network fresh without kademlia node discovery and random lookups.
+
+## Design
+
+### Nodes reach out to satellites that they want to work with
+- The satellites are listed in the trust package
+- Nodes should communicate with satellites directly rather than using kademlia to traverse the network to find the address of a given ID.
+- Storage nodes should notify satellites when they start up, wait a random amount of time (to add jitter 
```

Comment:
What is the scenario where storage node startup becomes a thundering herd?  Just in testing?

Reply:
@jtolds ?

Reply:
@wthorp Ok so it turns out watchtower updates nodes all at once. This means that all those nodes will try to ping the satellite at once when they come online, and therefore every hour. We need to add some jitter to the autoupdate system, but for now the jitter on start up pings should help prevent a thundering herd.